packages/dds/merge-tree/src/segmentGroupCollection.ts

```diff
@@ -21,9 +21,16 @@ export class SegmentGroupCollection {
         return this.segmentGroups.empty();
     }
 
-    public enqueue(segmentGroup: SegmentGroup) {
+    public enqueue(segmentGroup: SegmentGroup, sourceSegment?: ISegment) {
         this.segmentGroups.enqueue(segmentGroup);
         segmentGroup.segments.push(this.segment);
+        if (segmentGroup.previousProps && sourceSegment) {
```

Comment:
i'm confused how this works. segment groups can have multiple segments, and those segments can have different properties